packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -245,20 +268,6 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
           }
         }
       },
-
-      exit(path, state) {
-        if (
-          state.get("@babel/plugin-react-jsx/runtime") === "classic" &&
-          state.get("@babel/plugin-react-jsx/pragmaSet") &&
-          state.get("@babel/plugin-react-jsx/usedFragment") &&
-          !state.get("@babel/plugin-react-jsx/pragmaFragSet")
-        ) {
-          throw new Error(
```

Comment:
I don't think this makes much sense - it's overly restrictive. The old transform had no such validation and TS also has never thrown for this situation:
https://www.typescriptlang.org/play?ts=4.1.0-beta#code/PQKhAIAECsGcA9x0SYAoNBuAPAEwJYBuAfNsASRjqcMUA

Reply:
Can you just comment out this code for now?

I'm not 100% sure that we should remove this check (well, we should now for backward compat, but we might re-introduce it in Babel 8). The question is: should we prioritize for React libraries (e.g. Emotion) or for other "frameworks" (e.g. Preact)?

In the first case it's better not to throw because it should "work by default", while in the second case it's better to "prevent a mistake by default".

Reply:
> The question is: should we prioritize for React libraries (e.g. Emotion) or for other "frameworks" (e.g. Preact)?

Yeah, definitely a concern and I don't have a perfect answer for that. My main motivation factors here are that not throwing in this case is:
- backward-compatible
- matching how TS behaves

Using pragmas is not a very common use case so I would be inclined to favor backward compatibility, usually, people should really have those things configured globally per project in a config file. Pragmas are mostly useful in configuration-constrained tools.